test/unit/cmap/auth/mongodb_oidc/token_entry_cache.test.ts

```diff
@@ -0,0 +1,143 @@
+import { expect } from 'chai';
+
+import { TokenEntryCache } from '../../../../mongodb';
+
+describe('TokenEntryCache', function () {
+  const tokenResult = {
+    accessToken: 'test',
+    expiresInSeconds: 100
+  };
+
+  const serverResult = {
+    clientId: 1
+  };
+
+  describe('#addEntry', function () {
+    context('when expiresInSeconds is provided', function () {
+      const cache = new TokenEntryCache();
+      let entry;
+
+      before(function () {
+        cache.addEntry('localhost', 'user', tokenResult, serverResult);
+        entry = cache.getEntry('localhost', 'user');
+      });
+
+      it('sets the username in the key', function () {
+        expect(cache.entries.has('localhost-user')).to.be.true;
+      });
+
+      it('adds the token result', function () {
+        expect(entry.tokenResult).to.deep.equal(tokenResult);
+      });
+
+      it('adds the server result', function () {
+        expect(entry.serverResult).to.deep.equal(serverResult);
+      });
+
+      it('creates an expiration', function () {
+        expect(entry.expiration).to.be.above(Date.now());
+      });
+    });
+
+    context('when expiresInSeconds is not provided', function () {
+      const cache = new TokenEntryCache();
+      let entry;
+
+      const expiredResult = { accessToken: 'test' };
+
+      before(function () {
+        cache.addEntry('localhost', 'user', expiredResult, serverResult);
+        entry = cache.getEntry('localhost', 'user');
+      });
+
+      it('sets an immediate expiration', function () {
+        expect(entry.expiration).to.be.below(Date.now() + 1);
+      });
+    });
+
+    context('when expiresInSeconds is null', function () {
+      const cache = new TokenEntryCache();
+      let entry;
+
+      const expiredResult = {
+        accessToken: 'test',
+        expiresInSeconds: null
+      };
+
+      before(function () {
+        cache.addEntry('localhost', 'user', expiredResult, serverResult);
+        entry = cache.getEntry('localhost', 'user');
+      });
+
+      it('sets an immediate expiration', function () {
+        expect(entry.expiration).to.be.below(Date.now() + 1);
```

Comment:
is the `+1` necessary?

Reply:
Changed to test `at.most(Date.now())` - the +1 was there to be defensive about code running fast enough to where `Date.now()` could be the same for both the expiration and the call in the assertion.

Reply:
we lost this test for explicit `null`, I added a comment in my following review